projects/kit/components/routable-dialog/routable-dialog.component.ts

```diff
@@ -1,40 +1,51 @@
-import {ChangeDetectionStrategy, Component, inject, INJECTOR} from '@angular/core';
+import {ChangeDetectionStrategy, Component, inject, INJECTOR, Type} from '@angular/core';
 import {ActivatedRoute, Router} from '@angular/router';
 import {TuiDestroyService} from '@taiga-ui/cdk';
 import {TuiDialogService} from '@taiga-ui/core';
 import {PolymorpheusComponent} from '@tinkoff/ng-polymorpheus';
 import {takeUntil} from 'rxjs';
 
 @Component({
+    standalone: true,
     selector: 'tui-routable-dialog',
     template: '',
     changeDetection: ChangeDetectionStrategy.OnPush,
     providers: [TuiDestroyService],
 })
-export class TuiRoutableDialogComponent {
+export default class TuiRoutableDialogComponent {
     private readonly route = inject(ActivatedRoute);
     private readonly router = inject(Router);
+    private readonly injector = inject(INJECTOR);
     private readonly initialUrl = this.router.url;
+    private readonly dialog = inject(TuiDialogService);
+    private readonly destroy$ = inject(TuiDestroyService, {self: true});
 
     constructor() {
-        inject(TuiDialogService)
-            .open(
-                new PolymorpheusComponent(
-                    this.route.snapshot.data['dialog'],
-                    inject(INJECTOR),
-                ),
-                this.route.snapshot.data['dialogOptions'],
-            )
-            .pipe(takeUntil(inject(TuiDestroyService, {self: true})))
-            .subscribe({
-                complete: () => this.onDialogClosing(),
-            });
+        const dialog = this.route.snapshot.data['dialog'];
+
+        if (isFunction(dialog)) {
+            void Promise.resolve(dialog()).then(module =>
+                this.openDialog(module.default ?? module),
+            );
+        } else {
+            this.openDialog(dialog);
```

Comment:
Could we just start the stream using `from` ?

Something like this:

```ts
from(isFunction(dialog) ? dialog().then(m => m.default ?? m) : dialog)
    .pipe(
        switchMap(dialog => this.dialogService.open(dialog)),
        // [...]
        takeUntil(this.destroy$),
    )
    .subscribe();
```

Reply:
fixed